server/service/appconfig.go

```diff
@@ -628,13 +630,31 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 	return obfuscatedAppConfig, nil
 }
 
+func (svc *Service) HasCustomSetupAssistantConfigurationWebURL(ctx context.Context, teamID *uint) (bool, error) {
```

Comment:
Given that this is an exported Service method, I'd add a check that requires that authorization has already been done at the start, and return in error if that's not the case. Just to prevent misuse of this method in other places.

Reply:
ah, great catch. Thanks! Do you know off the top of your head if we already have methods to do that? I'm blanking, but I remember there's a way.

maybe using?

https://github.com/fleetdm/fleet/blob/6bd8263d4fc4c7f716eefd4be8372d87b047171f/server/contexts/authz/authz.go#L58

Reply:
Yes, that's exactly it (https://github.com/fleetdm/fleet/blob/main/server/contexts/authz/authz.go#L58-L62)!